src/test/groovy/synacy/gradprogram/spock/exercise/OrderingServiceSpec.groovy

```diff
@@ -0,0 +1,66 @@
+package synacy.gradprogram.spock.exercise
+
+import com.synacy.gradprogram.spock.exercise.CancelOrderRequest
+import com.synacy.gradprogram.spock.exercise.Order
+import com.synacy.gradprogram.spock.exercise.OrderRepository
+import com.synacy.gradprogram.spock.exercise.OrderStatus
+import com.synacy.gradprogram.spock.exercise.OrderingService
+import com.synacy.gradprogram.spock.exercise.RefundService
+import com.synacy.gradprogram.spock.exercise.UnableToCancelException
+import spock.lang.Specification
+
+class OrderingServiceSpec extends Specification {
+    OrderingService orderingService
+
+    OrderRepository orderRepository = Mock(OrderRepository)
+    RefundService refundService = Mock(RefundService)
+
+    void setup() {
+        orderingService = new OrderingService(orderRepository, refundService)
+    }
+
+    def "cancelOrder should throw UnableToCancelException when OrderStatus is CANCELLED or DELIVERED"() {
+        given:
+        def cancelOrderRequest = Mock(CancelOrderRequest)
+        cancelOrderRequest.getOrderId() >> orderUUID
+
+        Order order = Mock(Order)
+        order.getStatus() >> orderStatus
+
+        this.orderRepository.fetchOrderById(orderUUID) >> Optional.ofNullable(order)
+
+        when:
+        orderingService.cancelOrder(cancelOrderRequest)
+
+        then:
+        thrown(UnableToCancelException)
+
+        where:
+        orderStatus | orderUUID
+        OrderStatus.CANCELLED | UUID.randomUUID()
+        OrderStatus.DELIVERED | UUID.randomUUID()
+    }
+
+    def "cancelOrder should cancel PENDING and FOR_DELIVERY orders and create a refund request"() {
+        given:
+        def cancelOrderRequest = Mock(CancelOrderRequest)
+        cancelOrderRequest.getOrderId() >> orderUUID
+
+        Order order = Mock(Order)
+        order.getStatus() >> orderStatus
+
+        this.orderRepository.fetchOrderById(orderUUID) >> Optional.ofNullable(order)
+
+        when:
+        orderingService.cancelOrder(cancelOrderRequest)
+
+        then:
+        1 * order.setStatus(OrderStatus.CANCELLED)
```

Comment:
No need to check the setting of the status, we should just test what the output or what the method is actually doing :+1: The checking of status will be checked if we add the test that checks that we are saving the updated order.